pkg/pillar/cmd/baseosmgr/handledownload.go

```diff
@@ -98,32 +98,32 @@ func checkContentTreeStatus(ctx *baseOsMgrContext,
 
 // Note: can not do this in volumemgr since it is triggered by Activate=true
 func installDownloadedObjects(ctx *baseOsMgrContext, uuidStr, finalObjDir string,
-	status *[]types.ContentTreeStatus) bool {
+	status *[]types.ContentTreeStatus) (bool, bool, error) {
 
-	ret := true
+	var (
+		changed bool
+		proceed bool
+		err     error
+	)
 	log.Infof("installDownloadedObjects(%s)", uuidStr)
 
 	for i := range *status {
 		ctsPtr := &(*status)[i]
 
 		if ctsPtr.State == types.LOADED {
-			ready, err := installDownloadedObject(ctx, ctsPtr.ContentID,
+			changed, err = installDownloadedObject(ctx, ctsPtr.ContentID,
 				finalObjDir, ctsPtr)
 			if err != nil {
 				log.Error(err)
-			}
-			if !ready {
-				ret = false
+				return changed, proceed, err
 			}
 		}
-		// if something is still not installed, mark accordingly
-		if ctsPtr.State != types.INSTALLED {
-			ret = false
+		if ctsPtr.State == types.INSTALLED {
```

Comment:
It needs to get to LOADED state before proceeding AFAIU

Reply:
That check is already there inside the underlying function. So, we don't need to check it here.

Reply:
I was confused; INSTALLED > LOADED. But there are some comments which say VERIFIED to INSTALLED which should say LOADED to INSTALLED.